gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/FlowLaunchHandler.java

```diff
@@ -233,7 +234,7 @@ protected JobDetailImpl createJobDetailForReminderEvent(JobKey originalKey, JobK
       throws SchedulerException {
     JobDetailImpl jobDetail = (JobDetailImpl) this.schedulerService.getScheduler().getJobDetail(originalKey);
     jobDetail.setKey(reminderKey);
-    JobDataMap jobDataMap = jobDetail.getJobDataMap();
+    JobDataMap jobDataMap = deepCopyJobDataMap(jobDetail.getJobDataMap());
     jobDataMap = updatePropsInJobDataMap(jobDataMap, status, schedulerMaxBackoffMillis);
```

Comment:
updatePropsInJobDataMap already updates jobDataMap. So we just need to do `updatePropsInJobDataMap(jobDataMap, status, schedulerMaxBackoffMillis);` intellij flags this line